lib/rubocop/cop/layout/space_inside_array_literal_brackets.rb

```diff
@@ -88,8 +88,9 @@ def right_array_bracket(node)
         end
 
         def empty_brackets?(left_bracket_token, right_bracket_token)
-          processed_source.tokens.index(left_bracket_token) ==
-            processed_source.tokens.index(right_bracket_token) - 1
+          left_index = processed_source.tokens.index(left_bracket_token)
+          right_index = processed_source.tokens.index(right_bracket_token)
+          right_index && left_index == right_index - 1
```

Comment:
Without this change, the `rake spec` and `rake internal_investigation` still pass.  But, `rake_internal_investigation` would throw errors in a few spec files. The issue was the `right_bracket_token` index returning `nil`. This was mostly in the `or_assignment_spec` and only when a single line array contained a heredoc declaration with trailing methods. Example: https://github.com/bbatsov/rubocop/blob/master/spec/rubocop/cop/style/or_assignment_spec.rb#L237 .

Since `empty_brackets?` reports whether brackets are empty and this scenario is a definite false, this seems fine to me.